### PR TITLE
fix(vite): extract styles for shared chunks

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

- [x] resolves https://github.com/nuxt/nuxt/issues/25420
- [x] resolves https://github.com/nuxt/nuxt/issues/25451
- [x] resolves https://github.com/nuxt/nuxt/issues/22507
- [x] resolves https://github.com/nuxt/nuxt/issues/24480
- [x] resolves https://github.com/nuxt/nuxt/issues/25330

closes https://github.com/nuxt/nuxt/pull/25394

### ❓ Type of change
- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

We weren't extracting styles for chunks without a `facadeModuleId`, but it's safe to do so for individual module ids. This is a PR that simply adds these additional module ids to be extracted, as long as they are Vue files (which are the ones tracked in ssr context).

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] added tests

### CODE REVIEW

**Review by huang-julien — APPROVED** [review]

File: test/basic.test.ts [hunk 1]
```diff
@@ -1950,6 +1952,11 @@ describe('component islands', () => {
       expect(result.head).toMatchInlineSnapshot(`
         {
           "link": [
+            {
+              "href": "/_nuxt/components/SharedComponent.vue?vue&type=style&index=0&scoped=3ee84738&lang.css",
+              "key": "island-link",
+              "rel": "stylesheet",
+            },
```
Comment (danielroe) [inline test/basic.test.ts:1959]: This is not a product of changes in the code; this is vite behaviour for adding a shared component (also on main).

**Comment by danielroe** [comment]
Seems that this PR also resolves issue in https://github.com/nuxt/nuxt/pull/25394 (added regression test from that PR here as well).

File: packages/vite/src/utils/index.ts [hunk 2]
```diff
@@ -11,6 +13,39 @@ export function isCSS (file: string) {
   return IS_CSS_RE.test(file)
 }
 
+export function isVue (id: string, opts: { type?: Array<'template' | 'script' | 'style'> } = {}) {
```
Comment (huang-julien) [inline packages/vite/src/utils/index.ts:16]: Can we use directly the util from the core ?

Comment (huang-julien) [reply packages/vite/src/utils/index.ts:16]: `core/utils/plugins`

Comment (danielroe) [reply packages/vite/src/utils/index.ts:16]: It's a different package and I'm reluctant to expose the export. But maybe we can inline it instead

**Review by huang-julien — APPROVED** [review]

**Pull request merged**